README.md

```diff
@@ -35,54 +34,44 @@ This makes it much easier to contribute new packages packages, since there's no
 # Detailed design
 [design]: #detailed-design
 
-This RFC establishes the convention of `pkgs/unit/${toLower (substring 0 4 name)}/${name}` "unit" directories for the definitions of the Nix packages `pkgs.${name}` in nixpkgs.
-The `pkg-fun.nix` files in all unit directories are automatically discovered, called using `pkgs.callPackage` and added to the `pkgs` set.
+This RFC establishes the standard of using `pkgs/unit/${shard}/${name}` "unit" directories for the definitions of the Nix packages `pkgs.${name}` in nixpkgs, where `shard = toLower (substring 0 2 name)`.
+All unit directories are automatically discovered and incorporated into the `pkgs` set using `pkgs.${name} = pkgs.callPackage pkgs/unit/${shard}/${name}/pkg-fun.nix { }`.
 
-These requirements will be checked using CI:
-1. The `pkgs/unit` directory must only contain unit directories, and only in subdirectories of the form `${substring 0 4 name}/${name}`.
-2. <a id="req-ref-out"/> Files inside a unit directory must not reference files outside that unit directory.
-3. <a id="req-ref-in"/> Files outside a unit directory must not reference files inside a unit directory, except for definitions of attributes in `all-packages.nix` and the auto-calling logic.
-4. The definition of a package in the unit directory is the one `pkgs.<name>` points to.
-5. To avoid problems with merges, if a package attribute is defined by a unit directory, an attribute of the same name in `pkgs/top-level/all-packages.nix` (or `pkgs/top-level/aliases.nix`) must not redefine it in terms of something other than the unit.
+These requirements will be checked by CI:
+1. Structure: The `pkgs/unit` directory must only contain unit directories, and only in subdirectories of the form `${shard}/${name}`.
+  Each unit directory must contain at least a `pkg-fun.nix` file, but may contain arbitrary other files and directories.
+2. Semantics: Calling `pkgs.callPackage pkgs/unit/${shard}/${name}/pkg-fun.nix {}` must return a derivation that can be built directly with `nix-build`.
+3. Context-freedom: Files inside a unit directory must not reference files outside that unit directory, and the other way around.
+  The only exception is the next requirement.
```

Comment:
```suggestion
  The only exception is the next requirement. Dependencies from outside the unit only come in through the `pkg-fun.nix` function arguments.
```

A clarification for those who may be puzzled by this supposed isolation.

Reply:
sgtm. You've already titled them, so maybe turn them into subsections?

Also in the first subitem you just wrote: not just moving around, but other changes as well; use the public interface of those packages. EDIT: you explain it better the second time.

In the second: `pkgs.${name}` or `callPackage` injection.

Reply:
Good idea, I added sections now, I can't fully parse your reply, but I changed it a bit to emphasize the stable interface.